### ISSUE TITLE
improve cloud API

### ISSUE THREAD

**Issue by thomasgilgenast** [sclabs/scootbot2.0#4]
some things that would be nice include:

### pass through the virtual hostname to the container via an environment variable

for example, a Django app needs to know what its own hostname is so it can put that hostname in the `ALLOWED_HOSTS` whitelist config variable

for the higlass container we passed `-e SITE_URL=higlass.gilgi.org`, but is there a standard environment variable for this that all containers can agree on?

### allow subdomains to request a volume

our current "update+redeploy" pattern (which is correct afaik) is to tear down the container, pull a fresh one, and then restart the container

if the container wants to persist anything between updates, it would have to dump it to a shared volume

we could allow a `-v /container/dir` flag on `!cloud deploy` which would get translated to `-v ~/cloud/<subdomain>:/container/dir` in the `docker run` call

### allow `--expose` to be passed at deploy time

right now, the container's `Dockerfile` must specify an `EXPOSE` port, otherwise the cloud system has no way of knowing how to connect to the container

in other words, if people are used to starting their containers with `-p 80:80` and never formally exposed the port, this will not fly in the cloud

a solution could be to add a `-e <port>` flag on `!cloud deploy` which would get translated to `--expose=<port>` in the `docker run` call

**Comment by thomasgilgenast** [comment]
as of 2d7160a we allow volumes and environment variables to be passed through by adding them to the google sheet that stores the subdomain registration information. one major advantage of this approach is that you don't need to re-request the volume or re-specify the environment variables you depend on on every deploy. the major remaining shortcoming is that there is no way to request these via the `!cloud register` command - they have to be manually added to the sheet
